src/Compilers/CSharp/Test/Semantic/Semantics/StackAllocSpanExpressionsTests.cs

```diff
@@ -118,39 +112,33 @@ public void Method()
             Assert.Equal(SyntaxKind.CastExpression, obj1.Initializer.Value.Kind());
 
             var obj1Value = model.GetSemanticInfoSummary(((CastExpressionSyntax)obj1.Initializer.Value).Expression);
-            Assert.Null(obj1Value.Type);
+            Assert.Equal("Span", obj1Value.Type.Name);
             Assert.Equal("Span", obj1Value.ConvertedType.Name);
-            Assert.Equal(ConversionKind.StackAllocToSpanType, obj1Value.ImplicitConversion.Kind);
-            Assert.Equal(ConversionKind.Identity, obj1Value.ImplicitConversion.UnderlyingConversions.Single().Kind);
+            Assert.Equal(ConversionKind.Identity, obj1Value.ImplicitConversion.Kind);
 
             var obj2 = variables.ElementAt(1);
             Assert.Equal("obj2", obj2.Identifier.Text);
 
             var obj2Value = model.GetSemanticInfoSummary(obj2.Initializer.Value);
-            Assert.Null(obj2Value.Type);
-            Assert.True(obj2Value.ConvertedType is PointerTypeSymbol);
-            Assert.Equal("Int32", ((PointerTypeSymbol)obj2Value.ConvertedType).PointedAtType.Name);
-            Assert.Equal(ConversionKind.StackAllocToPointerType, obj2Value.ImplicitConversion.Kind);
-            Assert.Equal(ConversionKind.Identity, obj2Value.ImplicitConversion.UnderlyingConversions.Single().Kind);
+            Assert.Equal(SpecialType.System_Int32, ((PointerTypeSymbol)obj2Value.Type).PointedAtType.SpecialType);
```

Comment:
We need to confirm with what the spec will say about this. 
The Type might need to be <null> here and conversionKind should be StackAllocToPointerType

Similar would apply to the Span case.

Let's not change this now. It will be a simple change in semantic model once we confirm what needs to be reported for this per spec.

Reply:
There will be no conversions in the tree now after merging this PR. Everything will be a bound converted stackalloc node.

---
In reply to: [135353665](https://github.com/dotnet/roslyn/pull/21746#discussion_r135353665) [](ancestors = 135353665)